dialog/text.go

```diff
@@ -0,0 +1,47 @@
+package dialog
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+const (
+	// absolute max width of text dialogs
+	// (prevent them from looking unnaturally large on desktop)
+	maxTextDialogAbsoluteWidth float32 = 600
+
+	// max width of text dialogs as a percentage of the current window width
+	maxTextDialogWinPcntWidth float32 = .9
+)
+
+func newTextDialog(title, message string, icon fyne.Resource, parent fyne.Window) *dialog {
+	d := &dialog{
+		title:   title,
+		icon:    icon,
+		parent:  parent,
+		content: newCenterWrappedLabel(message),
+	}
+	d.beforeShowHook = createBeforeShowHook(d, message)
+
+	return d
+}
+
+// returns a beforeShowHook that sets the desired width of the dialog to the min of:
+// - width needed to show message without wrapping
+// - maxTextDialogAbsoluteWidth
+// - current window width * maxTextDialogWinPcntWidth
+func createBeforeShowHook(d *dialog, message string) func() {
+	noWrapWidth := widget.NewLabel(message).MinSize().Width + padWidth + theme.Padding()*2
```

Comment:
It is better to use `fyne.MeasureText` rather than allocating a temp label.

Reply:
It resulted in a different (slightly) size, and failing tests. But if we want to update all the rendering tests (I don't know how to regenerate the XMLs), I suppose we could do that

Reply:
Created issue #4648 and added an explanatory comment to the code, referencing it